File: PHP/api_call.php
<?php
/**
 * TME API - code snippet.
 *
 * More info at: https://developers.tme.eu
 */
$token = '<put_your_token_here>';
$secret_key = '<put_your_secret_here>';
$params = array(
  'SymbolList' => array('1N4007'),
  'Country' => 'PL',
  'Currency' => 'PLN',
  'Language' => 'PL',
);

$response = api_call('Products/GetPrices', $params, $token, $secret_key);
$array = json_decode($response, true);

print_r($array);


//---------------------------------------------------------
function api_call($action, $params, $token, $secret_key)
{
    $api_url = 'https://api.tme.eu/' . $action . '.json';
    $curl = curl_init();

    // calculate HMAC-SHA1 signature
    $params['Token'] = $token;
    ksort($params);

    // In PHP 5.4 http_build_query() offers enc_type parameter (PHP_QUERY_RFC3986) which replaces bellow snippet.
    $encoded_params = str_replace(
        array('+', '%7E'),
        array('%20', '~'),
        http_build_query($params)
    );
    $signature_base = 'POST' . '&' . rawurlencode($api_url) . '&' . rawurlencode($encoded_params);
    $api_signature  = base64_encode(hash_hmac('sha1', $signature_base, $secret_key, true));

    $params['ApiSignature'] = $api_signature;
    //--

    curl_setopt($curl, CURLOPT_URL, $api_url);
    curl_setopt($curl, CURLOPT_POST, true);
    curl_setopt($curl, CURLOPT_POSTFIELDS, http_build_query($params));
    curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, true);

    $json_data = curl_exec($curl);
    curl_close($curl);
    
    return $json_data;
}

File: PYTHON/api_call.py
# TME API - example usage with Python
# More info at: https://developers.tme.eu

import collections, urllib, base64, hmac, hashlib, urllib2, json

def product_import_tme():
    # /product/product_import_tme/

    token = '<TOKEN>'
    app_secret = '<APP SECRET>'

    params = {
	'SymbolList[0]' : '1N4002',
        'SymbolList[1]' : '1N4007',
        'Country': 'PL',
        'Currency': 'PLN',
        'Language': 'PL',
    }

    response = api_call('Products/GetPrices', params, token, app_secret, True);
    response = json.loads(response)
    print response

def api_call(action, params, token, app_secret, show_header=False):
    api_url = 'https://api.tme.eu/' + action + '.json';
    params['Token'] = token;

    params = collections.OrderedDict(sorted(params.items()))

    encoded_params = urllib.urlencode(params, '')
    signature_base = 'POST' + '&' + urllib.quote(api_url, '') + '&' + urllib.quote(encoded_params, '')

    api_signature = base64.encodestring(hmac.new(app_secret, signature_base, hashlib.sha1).digest()).rstrip()
    params['ApiSignature'] = api_signature

    opts = {'http' :
               {
               'method' : 'POST',
               'header' : 'Content-type: application/x-www-form-urlencoded',
               'content' : urllib.urlencode(params)
               }
            }

    http_header = {
        "Content-type": "application/x-www-form-urlencoded",
        }
    # create your HTTP request
    req = urllib2.Request(api_url, urllib.urlencode(params), http_header)
    # submit your request
    res = urllib2.urlopen(req)
    html = res.read()
    return html

print product_import_tme();
